compiler/src/main/java/com/xfinity/blueprint_compiler/CodeGenerator.java

```diff
@@ -345,14 +349,43 @@ private MethodSpec getSetVisibilityMethodSpec(String childName, String childGett
         ParameterSpec visibilityParam =
                 ParameterSpec.builder(INT, "visibility").build();
 
+        AnnotationSpec deprecatedAnnotation = AnnotationSpec.builder(ClassName.get("java.lang", "Deprecated")).build();
+
         return MethodSpec.methodBuilder("set" + childName + "Visibility")
+                          .addJavadoc("@deprecated.  Use show$L(), hide$L(), or make$LInvisible() instead", childName,
+                                      childName, childName)
+                          .addAnnotation(deprecatedAnnotation)
                           .addModifiers(PUBLIC)
                           .addParameter(visibilityParam)
                           .returns(VOID)
                           .addStatement("viewHolder." + childGetterName + ".setVisibility(visibility)")
                           .build();
     }
 
+    private MethodSpec getShowMethodSpec(String childName, String childGetterName) {
+        return MethodSpec.methodBuilder("show" + childName)
+                         .addModifiers(PUBLIC)
+                         .returns(VOID)
+                         .addStatement("viewHolder." + childGetterName + ".setVisibility(android.view.View.VISIBLE)")
+                         .build();
+    }
+
+    private MethodSpec getHideMethodSpec(String childName, String childGetterName) {
+        return MethodSpec.methodBuilder("hide" + childName)
```

Comment:
Hide feels a little ambiguous regarding gone/invisible. Should we just have makeChildNameGone/Visible/Invisible?

Reply:
I had that originally, but i personally like the ambiguousness of it, since it masks all knowledge of the Android visibility system.  The makeInvisible function is really just for completeness.

@mrtvrgn thoughts?

Reply:
For what it's worth, if you want to be ambiguous and mask the existence of Android, I'd be fine just dropping the invisible function altogether. That said, we are building an Android-specific library to support RecyclerViews, per the library description. I'm not sure there's a lot of point to masking Android specifics in this.

Reply:
I prefer this way. We are frequently using show/hide functionality and I believe it is fine like this. As @dapp mentioned, it is for completeness.